File: tests/go/39_fref_param.go
package main

/*
Case 39: Go 跨作用域 — 函数参数传递
将 os/exec.Command 作为函数值参数传入
预期: 检出 CVI-8000
*/
import (
	"os"
	"os/exec"
)

func runWith(execFn func(name string, arg ...string) *exec.Cmd, arg string) {
	cmd := execFn("sh", "-c", arg)
	cmd.Run()
}

func main() {
	if len(os.Args) < 2 {
		return
	}
	runWith(exec.Command, os.Args[1])
}

File: tests/python/39_fref_param.py
"""
Case 39: Python 跨作用域 — 函数参数传递函数引用
将 os.system 作为参数传入 executor，executor 通过参数调用
预期: 检出 CVI-7000
"""
import os
import sys

def executor(op, arg):
    op(arg)

if __name__ == '__main__':
    user_input = sys.argv[1]
    executor(os.system, user_input)
